src/Effect/Exception.ss

```diff
@@ -0,0 +1,76 @@
+;; -*- mode: scheme -*-
+
+(library (Effect.Exception foreign)
+  (export showErrorImpl
+          error
+          errorWithCause
+          message
+          name
+          stackImpl
+          throwException
+          catchException)
+  (import (only (rnrs base) define lambda if for-each)
+          (only (rnrs conditions)
+            condition
+            condition-message
+            simple-conditions
+            make-message-condition
+            condition?
+            message-condition?)
+          (only (rnrs io ports) call-with-string-output-port)
+          (only (rnrs io simple) newline)
+          (only (rnrs exceptions) with-exception-handler raise-continuable)
+          (only (chezscheme) format call/cc display-condition))
+
+  (define showErrorImpl
+    (lambda (err)
+      (if (condition? err)
+        (call-with-string-output-port
+          (lambda (p)
+            (for-each
+              (lambda (e) (display-condition e p) (newline p) (newline p))
+              (simple-conditions err))))
```

Comment:
Actually now that I'm playing with this more in the repl I learned that `display-condition` already does some smarts around the compound condition:

```
> (display-condition (condition (make-warning) (make-message-condition "boom")))
Warning: boom
> (display-condition (condition (make-message-condition "boom")))
Exception: boom
```

So if you add a `make-warning` condition to the compound condition, then it will display it with "Warning". So it seems we could just simply use `display-condition` and not loop over the simple-conditions. Though this means the "cause" might be lost a bit if `display-condition` doesn't always display all the conditions. I think this is fine though, we can improve it later, but I'd rather keep this very simple for now.

Reply:
Yes it always discards the "cause". My thought process here was that `showErrorImpl` shows all of the stack of exceptions, and `message` extracts only the recent-most one, discarding the causes (if any). But I don't expect people to depend much on the `show` semantics anyways so I don't mind this change.